src/App/Pages/Accounts/LoginPasswordlessViewModel.cs

```diff
@@ -0,0 +1,157 @@
+                        nameof(LogInAttemptByLabel),
+                        nameof(TimeOfRequestText),
+                    });
+            }
+        }
+
+        private FormattedString CreateFingerprintPhrase(string fingerprintPhrase)
+        {
+            try
+            {
+                return fingerprintPhrase?.Split(new[] { Constants.FingerprintPhraseSeparator }, StringSplitOptions.RemoveEmptyEntries).Aggregate(new FormattedString(), (fs, fingerprint) =>
+                {
+                    if (fs.Spans.Any())
+                    {
+                        fs.Spans.Add(new Span
+                        {
+                            Text = Constants.FingerprintPhraseSeparator,
+                            TextColor = ThemeManager.GetResourceColor("DangerColor")
+                        });
+                    }
+
+                    fs.Spans.Add(new Span
+                    {
+                        Text = fingerprint
+                    });
+
+                    return fs;
+                });
+            }
+            catch (Exception ex)
+            {
+                _logger.Exception(ex);
+                return null;
+            }
+        }
+
+        private string CreateRequestDate(DateTime requestDate)
+        {
+            var minutesSinceRequest = requestDate.ToUniversalTime().Minute - DateTime.UtcNow.Minute;
+            if (minutesSinceRequest < 5)
+            {
+                return AppResources.JustNow;
+            }
+            if (minutesSinceRequest < 59)
+            {
+                return string.Format(AppResources.MinutesAgo, minutesSinceRequest);
+            }
+
+            return requestDate.ToShortTimeString();
+        }
+
+        private async Task AcceptRequestAsync()
+        {
+            await _deviceActionService.ShowLoadingAsync(AppResources.Loading);
+            var res = await _authService.LogInPasswordlessAcceptAsync();
+            await _deviceActionService.HideLoadingAsync();
+            await Page.Navigation.PopModalAsync();
+            _platformUtilsService.ShowToast("info", null, AppResources.LogInAccepted);
+        }
+
+        private async Task RejectRequestAsync()
+        {
+            await _deviceActionService.ShowLoadingAsync(AppResources.Loading);
+            var res = await _authService.LogInPasswordlessRejectAsync();
+            await _deviceActionService.HideLoadingAsync();
+            await Page.Navigation.PopModalAsync();
+            _platformUtilsService.ShowToast("info", null, AppResources.LogInDenied);
+        }
+
+        private void HandleServerErrorException(Exception ex)
+        {
+            Xamarin.Essentials.MainThread.BeginInvokeOnMainThread(async () =>
+            {
+                await _deviceActionService.HideLoadingAsync();
+                await _platformUtilsService.ShowDialogAsync(AppResources.GenericErrorMessage);
+            });
+            _logger.Exception(ex);
+        }
```

Comment:
I'd change the name, the exception could not be necessary from server, maybe the `ShowLoading` throws and that wouldn't be server error. And Server errors are handled by the `ApiException` if you wanted some custom behavior there.
Aside from that if an exception happens inside the invoke on main thread it will crash the application because of the context. If you need to do that, either add a try...catch internally or use the awaitable version of that which bubbles up the exception, i.e. `MainThread. InvokeOnMainThreadAsync(Func<Task>)`
Btw, is the invoke on main thread really needed here?

Reply:
Had to add the invoke on main thread because if you try to call the hide loading it will throw.